src/SessionSetup/RetrieveInstancesWidget.cpp

```diff
@@ -199,4 +219,44 @@ void RetrieveInstancesWidget::OnReloadButtonClicked() {
             });
 }
 
+std::optional<orbit_ggp::Project> RetrieveInstancesWidget::GetSelectedProject() {
+  std::optional<Project> project;
+  if (ui_->projectComboBox->currentData().canConvert<Project>()) {
+    project = ui_->projectComboBox->currentData().value<Project>();
+  }
+  return project;
+}
+
+void RetrieveInstancesWidget::OnProjectComboBoxCurrentIndexChanged() {
+  std::optional<Project> selected_project = GetSelectedProject();
+
+  // nothing changed, no loading required
+  if (selected_project == GetQSettingsProject()) return;
+
+  emit LoadingStarted();
+  retrieve_instances_->LoadInstances(selected_project, GetInstanceListScope())
+      .Then(main_thread_executor_.get(),
+            [this, selected_project](const ErrorMessageOr<QVector<Instance>>& load_result) {
+              // `this` still exists when this lambda is executed. This is enforced, because
+              // main_thread_executor_ is a member of `this`. When `this` is destroyed,
+              // main_thread_executor_ is destroyed and the lambda is not executed. Check
+              // Future::Then for details about the lambda not being executed.
+
+              if (load_result.has_value()) {
+                SetQSettingsProject(selected_project);
+              } else {
+                std::optional<Project> previously_selected = GetQSettingsProject();
+                if (previously_selected == std::nullopt) {
+                  ui_->projectComboBox->setCurrentIndex(0);
```

Comment:
Some comment about this 0? // Selecting the first at default is there isn't previous selection?

Reply:
done